### PR TITLE
convert to using jenkins user on worker

### PR DESCRIPTION
**Pull request by JoeNunnelley**
### Description

add both an ubuntu and jenkins user and convert the default user to jenkins

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] CHANGELOG has been updated
- [ ] Version has been bumped in metadata
- [ ] Commits have been squashed, as appropriate
- [ ] Commits include descriptive messages, subjects written in the imperative

### CODE REVIEW

File: recipes/_user.rb [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-  not_if "getent passwd #{worker_user}"
-end
+node['worker_users'].each do |worker_user|
+    user worker_user do
+        comment "user #{worker_user}"
+        home "/home/#{worker_user}"
+        shell '/bin/bash'
+        system true
+        action :create
+        not_if "getent passwd #{worker_user}"
```
Comment (RoboticCheese) [inline recipes/_user.rb]: Is there a purpose this guard is serving or is it redundant? Chef resources are idempotent--it won't try and fail to create a user that already exists.

File: recipes/_user.rb [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-  mode '0755'
-  action :create
-  not_if { File::exists?(user_home) }
+    directory "/home/#{worker_user}" do
+        user worker_user
+        owner worker_user
+        group worker_user
+        mode '0755'
+        action :create
+        not_if { File::exists?("/home/#{worker_user}") }
```
Comment (RoboticCheese) [inline recipes/_user.rb]: Ditto here. This is already how the directory resource works. This one would change the behavior in that Chef wouldn't fix the directory permissions if it were owned by another user, but... is that what you want it to do?

File: recipes/_chef.rb [hunk 3]
```diff
@@ -1,32 +1,35 @@
...
-  user node['worker_user']
-  group node['worker_user']
-end
 
-directory "#{user_home}/.kitchen" do
-  owner node['worker_user']
-  group node['worker_user']
-end
+node['worker_users'].each do |worker_user|
+    user_home = "/home/#{worker_user}"
```
Comment (RoboticCheese) [inline recipes/_chef.rb]: The spacing here is different from what this recipe had before and what other recipes in this cookbook have.

**Comment by JoeNunnelley** [comment]
retest this please

**Comment by JoeNunnelley** [comment]
retest this please

**Comment by JoeNunnelley** [comment]
retest this please

**Comment by JoeNunnelley** [comment]
retest this please

**Comment by JoeNunnelley** [comment]
retest this please

**Comment by JoeNunnelley** [comment]
retest this please

**Comment by JoeNunnelley** [comment]
merging so I can test AMI creation. converge  on jenkins still failing on 18.04 variant for reasons we don't fully understand. it converges fine locally.

**Pull request merged**